server/src/test/java/org/opensearch/rest/RestRequestTests.java

```diff
@@ -280,6 +282,37 @@ public void testRequiredContent() {
         assertEquals("unknown content type", e.getMessage());
     }
 
+    /*
+     * The test is added in 2.0 when the request parameter "cluster_manager_timeout" is introduced.
+     * Remove the test along with the removal of the non-inclusive terminology "master_timeout".
+     */
+    public void testValidateParamValuesAreEqual() {
+        FakeRestRequest request = new FakeRestRequest();
+        List<String> valueList = new ArrayList<>(Arrays.asList(null, "", "value1", "value2"));
+        String valueForKey1 = randomFrom(valueList);
+        String valueForKey2 = randomFrom(valueList);
+        request.params().put("key1", valueForKey1);
+        request.params().put("key2", valueForKey2);
+        try {
+            request.validateParamValuesAreEqual("key1", "key2");
+            assertTrue(
+                "Values of the 2 keys should be equal, or having a least 1 null value or empty String. Value of key1: "
+                    + valueForKey1
+                    + ". Value of key2: "
+                    + valueForKey2,
+                Strings.isNullOrEmpty(valueForKey1) || Strings.isNullOrEmpty(valueForKey2) || valueForKey1.equals(valueForKey2)
+            );
+        } catch (OpenSearchParseException e) {
```

Comment:
Can we please split this into two tests? 

We should deterministically test both cases - when the two are equal and for inequality - for all test runs rather than testing either code path through a random.

Reply:
Thanks for telling me the opinion to design test case! Updated along wit the above comment https://github.com/opensearch-project/OpenSearch/pull/2435/commits/02e68b59478fd9eff4e5c5c7f17560a702a8c1b5